### PR TITLE
Add `danger` theme color to icon-only buttons & allow color styling of TableRowActions

### LINKED ISSUE #2660: Change the color of table row action buttons
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [Modus style guidelines](https://modus.trimble.com/components/web/introduction/)

### Proposal

I wish to change the color of action row buttons within a table.
This would be done with a prop setting the icon color when defining the row actions.

### Motivation and context

This is to realize designs like this:
image

### What is the issue regarding ?

@trimble-oss/modus-web-components

### Priority

Medium

### What product/project are you using Modus Components for ?

Viewpoint Cascade

### What is your team/division name ?

Cascade

### Are you willing to contribute ?

Yes

### Are you using Modus Web Components in production ?

_No response_

### PR DESCRIPTION
**Pull request by Kai-Richardson**
## Description

This PR does several things:
* Extends the 'danger' color theme to icon-only buttons, as it's a valid `ButtonColor` for all of the other styles.
* Adds a prop to `TableRowAction` that allows you to specify the color of the icon shown.
* Cleans up the scss to reference the global trimble color variables.
* Fixes a bug with the Row Action page of the storybook where it was referencing an icon that did not exist.

Closes #2660

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

I worked in and verified behavior in Storybook.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx [hunk 1]
```diff
@@ -981,7 +982,8 @@ RowActions.args = {
 
     {
       id: '3',
-      icon: 'cancel',
+      icon: 'cancel_circle',
```
Comment (Kai-Richardson) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table.stories.tsx:985]: Here is the bug referenced in the PR body. There is no `cancel` icon. I chose to substutite `cancel_circle`, but there is also `cancel_square` if that's desired.

**Comment by cjwinsor** [comment]
@Kai-Richardson If you haven't already, can you make sure there is an Aha! submission for this so a review can be made?
https://modus.ideas.aha.io/ideas

This change is outside our available modus guidance so it will need to be reviewed by the design system team.

**Comment by Kai-Richardson** [comment]
> @Kai-Richardson If you haven't already, can you make sure there is an Aha! submission for this so a review can be made? [modus.ideas.aha.io/ideas](https://modus.ideas.aha.io/ideas)
> 
> This change is outside our available modus guidance so it will need to be reviewed by the design system team.

https://modus.ideas.aha.io/ideas/MOD-I-91

**Comment by cjwinsor** [comment]
@Kai-Richardson The request to add a danger styled icon only button was not approved by the modus design system.  They left feedback on the Aha! submission.  You can discuss with them in the comments section there.   For now I will close this PR.

**Pull request closed without merging**